hail/python/hailtop/utils/utils.py

```diff
@@ -33,15 +33,16 @@ def async_to_blocking(coro):
         thread_pool, lambda: fun(*args, **kwargs))
 
 
-async def bounded_gather(*pfs, parallelism=10, return_exceptions=False):
-    gatherer = AsyncThrottledGather(*pfs,
-                                    parallelism=parallelism,
-                                    return_exceptions=return_exceptions)
+async def bounded_gather(*pfs, **kwargs):
+    gatherer = AsyncThrottledGather(*pfs, **kwargs)
     return await gatherer.wait()
 
 
 class AsyncThrottledGather:
     def __init__(self, *pfs, parallelism=10, return_exceptions=False):
+        assert (return_exceptions is False or
+                return_exceptions is True or
+                return_exceptions > 0), return_exceptions
```

Comment:
I'm also sketical we have a use case for this intermediate collect N exceptions and then bail behavior.  I think we'll just want the extremes: raise the first exception encountered, or return any and call exceptions (which was what was implemented).

Reply:
I agree with Cotton.